tests/bluetooth/bsim_bt/bsim_test_audio/src/ias_client_test.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BT_IAS_CLIENT
+
+#include "bluetooth/services/ias.h"
+#include "common.h"
+
+extern enum bst_result_t bst_result;
+
+CREATE_FLAG(g_is_connected);
+CREATE_FLAG(g_service_discovered);
+CREATE_FLAG(g_alert_sent);
+
+static struct bt_conn *g_conn;
+
+static void discover_cb(struct bt_conn *conn, int err)
+{
+        if (err) {
+                FAIL("Failed to discover IAS (err %d)\n", err);
+                return;
+        }
+
+        printk("IAS discovered\n");
+	SET_FLAG(g_service_discovered);
+}
+
+static const struct bt_ias_client_cb ias_client_cb = {
+        .discover = discover_cb,
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	if (err > 0) {
+		char addr[BT_ADDR_LE_STR_LEN];
+
+		bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+		FAIL("Failed to connect to %s (err %u)\n", addr, err);
+		return;
+	}
+
+	g_conn = conn;
+	SET_FLAG(g_is_connected);
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void test_main(void)
+{
+	int err;
+
+	err = bt_enable(NULL);
+	if (err < 0) {
+		FAIL("Bluetooth discover failed (err %d)\n", err);
+		return;
+	}
+
+	printk("Bluetooth initialized\n");
+
+        err = bt_ias_client_cb_register(&ias_client_cb);
+        if (err < 0) {
+                FAIL("Failed to register callbacks (err %d)\n", err);
+                return;
+        }
+
+        err = bt_le_scan_start(BT_LE_SCAN_PASSIVE, device_found);
+	if (err < 0) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+		return;
+	}
+
+        printk("Scanning successfully started\n");
+
+	WAIT_FOR_COND(g_is_connected);
+        err = bt_ias_discover(g_conn);
+	if (err < 0) {
+		FAIL("Failed to discover IAS (err %d)\n", err);
+		return;
+	}
+
+        WAIT_FOR_COND(g_service_discovered);
+        WIAT_FOR_COND(g_alert_sent)
+
+        PASS("IAS main PASS\n");
+}
```

Comment:
Missing empty line at the end of file (most editors can be setup to add this automatically)